demo/Makefile

```diff
@@ -11,7 +11,7 @@ run: build
 build-snapshot:
 	docker build -t $(IMAGE):RELEASE .
 	mkdir -p snapshot-plugins
-	for p in $$(cat plugins.txt|perl -pe s/:.+//g; cat snapshot-only-plugins.txt); do echo looking for snapshot builds of $$p; for g in org/jenkins-ci/plugins org/jenkins-ci/plugins/workflow org/jenkins-ci/plugins/pipeline-stage-view; do if [ -f ~/.m2/repository/$$g/$$p/maven-metadata-local.xml ]; then cp -v $$(ls -1 ~/.m2/repository/$$g/$$p/*-SNAPSHOT/*.hpi | tail -1) snapshot-plugins/$$p.jpi; fi; done; done
+	for p in $$(cat snapshot-only-plugins.txt); do echo looking for snapshot builds of $$p; for g in org/jenkins-ci/plugins org/jenkins-ci/plugins/workflow org/jenkins-ci/plugins/pipeline-stage-view; do if [ -f ~/.m2/repository/$$g/$$p/maven-metadata-local.xml ]; then cp -v $$(ls -1 ~/.m2/repository/$$g/$$p/*-SNAPSHOT/*.hpi | tail -1) snapshot-plugins/$$p.jpi; fi; done; done
```

Comment:
The intention, and the original behavior of `workflow-plugin/demo/Dockerfile#run-snapshot`, was that you should not need to modify demo source files in order to pick up snapshots. If you are running a snapshot demo anyway, you are generally interested in knowing about any problems introduced by unreleased changes.

Now we could amend that to require you to list the plugins to be searched in a variable passed to `make`, e.g.

    make -C demo run-snapshot SNAPSHOT_PLUGINS='cloudbees-folder branch-api workflow-multibranch'

where `snapshot-only-plugins.txt` could be deleted in favor of temporary edits to the default value of this variable in the `Makefile`.

At any rate, this is better worked out in a separate PR.